app/models/tax_return.rb

```diff
@@ -155,6 +155,21 @@ def sign_spouse!(ip)
     true
   end
 
+  def assign!(assigned_user_id: nil, assigned_by: nil)
+    if update!(assigned_user_id: assigned_user_id)
```

Comment:
I think `update!` either updates or raises an exception; perhaps you want `if update`?

Given that we always expect it to succeed, I recommend removing the `if`, keeping the `update!`, and letting it crash if it needs to.